cpp/include/cudf/ast/expressions.hpp

```diff
@@ -38,13 +38,34 @@ class expression_parser;
  * Expressions inheriting from this class can accept parsers as visitors.
  */
 struct expression {
+  /**
+   * @pure @brief Accepts a visitor class.
```

Comment:
According to [the doxygen docs](https://www.doxygen.nl/manual/commands.html#cmdpure) the `pure` command 
> is intended for use only when the language does not support the concept of pure virtual methods natively

so why do we need it here?

Reply:
```suggestion
   * @brief Accepts a visitor class.
```